src/spec/cli.cr

```diff
@@ -0,0 +1,136 @@
+require "./formatter"
+require "option_parser"
+
+module Spec
+  # :nodoc:
+  class CLI
+    # :nodoc:
+    class Options
+      property formatters = Array(Spec::Formatter).new
+      property locations = Array({String, Int32}).new
+
+      property! default_formatter : Spec::Formatter
+      property! fail_fast : Bool
+      property! line : Int32
+      property! no_color : Bool
+      property! pattern : String
+      property! slowest : Int32
+    end
+
+    getter options : Options
+
+    private getter argv : Array(String)
```

Comment:
I disagree on the usefulness of it. private getters are used across the codebase and internals in similar way.

In the case of `argv`, not using it directly since it will require me to assign to local before I can retrieve `.first` (see `#prepare`).

Reply:
> it will require me to assign to local before I can retrieve .first

why?

Also I really don't care use whichever style you like.